pennylane/gradients/spsa_gradient.py

```diff
@@ -35,7 +36,7 @@
 from .general_shift_rules import generate_multishifted_tapes
 
 
-def _rademacher_sampler(indices, num_params, *args, seed=None):
+def _rademacher_sampler(indices, num_params, *args, rng):
```

Comment:
Since this function is not exposed externally and is guaranteed to receive a `np.random.Generator` in `spsa_grad()` I'd suggest making this a required (positional) argument. If we want to make it a kwarg, we'd have to make sure that it cannot be an integer, since this was the problem which initially lead to this PR, where the same integer was passed each time `_rademacher_sampler` was called.

Reply:
Ah, right. It's true, the function is private. Maybe we just want to make `rng` a required argument _always_, including in `coordinate_sampler` and in the descriptions in the docs? This way we avoid users copying and modifying the sampler and running into this trap? :thinking:

Reply:
I agree that this argument should be non-optional, since it's a sampler. But where would you change it in the docs? I only see the `coordinate_sampler` being defined in tests.

Reply:
Ah, sorry, could have been more clear on that. I marked two points where `rng` is explicitly called a keyword argument :)

Reply:
I see. Thanks!